src/PrestaShopBundle/Form/Admin/Type/DeltaQuantityType.php

```diff
@@ -58,6 +64,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'constraints' => [
                     new Type(['type' => 'numeric']),
                     new NotBlank(),
+                    new Range([
```

Comment:
This extra constraint is a good thing, however it won't be exact because this is only the delta quantity part which is added on the existing value. So depending on the initial value in DB this constraint won't block the value correctly (if quantity is already 1000 2147483600 would be considered valid even if it results in an out of bound modification).

However, this form type also includes the `quantity` field even if it's as a displayed text maybe we can put the constraint on the quantity field since it is computed and posted in the request, this will improve the check.

It would still leave the case when the DB was modified in the meantime but we don't need to push that far I'd say since the ObjectModel validation will detect this one during the CQRS command execution. However we should make sure that the controller is abal to display the exception correctly with a human readble message, this is handled in the controller `getErrorMessages` method which should handle the `ProductStockConstraintException` with the `INVALID_QUANTITY` error code.

This seems to be handled in the combination controller https://github.com/PrestaShop/PrestaShop/blob/bc18317950135ca1b9d4a82964c174e1b65dfc8f/src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php#L597-L601 but not in the product one which also uses this form type and the same validator and exceptions.

Reply:
Now if the quantity is bigger then max it just sets it to max. Maybe its better then showing error? Or should we stop it from updating if its bigger then max size?